src/components/tl/ucp/tl_ucp.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_TL_UCP_H_
+#define UCC_TL_UCP_H_
+#include "components/tl/ucc_tl.h"
+#include "components/tl/ucc_tl_log.h"
+#include <ucp/api/ucp.h>
+#include <ucs/memory/memory_type.h>
+
+typedef struct ucc_tl_ucp_iface {
+    ucc_tl_iface_t super;
+} ucc_tl_ucp_iface_t;
+/* Extern iface should follow the pattern: ucc_tl_<tl_name> */
+extern ucc_tl_ucp_iface_t ucc_tl_ucp;
+
+typedef struct ucc_tl_ucp_lib_config {
+    ucc_tl_lib_config_t super;
+} ucc_tl_ucp_lib_config_t;
+
+typedef struct ucc_tl_ucp_context_config {
+    ucc_tl_context_config_t super;
+    int                     test_param;
```

Comment:
?

Reply:
Doxygen style - does this mean it will be part of document build? Do we want to expose that to users? 
C -style comment is not enough? 
What do you want me to describe: overall CL/TL idea or details for all the fields and params in the structs?

Reply:
after discussion with Manju: will add short desc